### ISSUE TITLE
How to use pino.final with avvio

### ISSUE THREAD

**Issue by gyszalai** [pinojs/pino#490]
I have an avvio application with a few onClose handlers that run when the application is
stopped when the node process receives a SIGTERM signal.

I know that `pino.final()` is made for this purpose, but how can I use it in an onClose handler?

**Comment by jsumners** [comment]
Are you sure you don't want http://getpino.io/#/docs/api?id=flush ? Otherwise, I think the documentation is pretty clear http://getpino.io/#/docs/api?id=pino-final . What is unclear to you?

**Comment by gyszalai** [comment]
the flush() method is async so it doesn't work in my case, i.e. the last log entries are not flushed to the destination before the VM process exits.

Yes, the documentation for pino.final() is clear, but this call is for creating a callback that you can pass as an event handler. In my case I just need the finalLogger in my onClose handlers (quite a few of them). Actually I did the following to get get the final logger:
```js
const finalLogger = pino.final(logger, (_, finalLogger) => finalLogger)()
```
which is not really nice IMHO

**Comment by davidmarkclements** [comment]
We can adapt the api so that if a callback isn't supplied the final logger is returned - this would be a simple PR if you're up for contributing

**Comment by gyszalai** [comment]
Okay, I'll file a PR soon. Already working on it. It'll break the API though, as now `pino.final` it throws an Error if no handler is passed to it.

**Comment by davidmarkclements** [comment]
this isn't a breakage really, it's an api extension, since the PR has been merged and 5.3.0 has been released we're good to close

thanks @gyszalai  !
